File: src/packets/PacketC2SIntent.js
const PacketC2S = require("./PacketC2S");

/**
 * Client -> Server
 * ID = 0x01
 * Intent
 * Let's us know what the client plans on doing
 */
class PacketC2SIntent extends PacketC2S {
    parseData() {
        this._intent = this._buffer.readUInt8();
    }

    getIntent() {
        return this._intent;
    }

    getID() {
        return PacketC2SIntent.ID;
    }
}

PacketC2SIntent.ID = 0x01;

module.exports = PacketC2SIntent;

File: src/RNetPi.Core/Packets/PacketC2SIntent.cs
namespace RNetPi.Core.Packets;

/// <summary>
/// Client -> Server
/// ID = 0x01
/// Intent
/// Announces to the server the client's intent to connect
/// </summary>
public class PacketC2SIntent : PacketC2S
{
    public const byte ID = 0x01;

    public PacketC2SIntent(byte[] data) : base(data)
    {
    }

    public override byte GetID() => ID;

    protected override void ParseData()
    {
        // No data to parse for intent packet
    }
}